lantiq: ltq-atm: use reset_control_get_optional

neheb opened the pull request:
Based on the fact that the returns are commented out, this indicates that these are optional and as such should use the proper API.

Also simplifies error handling by using dev_err_probe and not having to handle -EPROBE_DEFER.

ping @hauke

neheb committed:
ltq-atm: propagate EPROBE_DEFER to probe

Instead of ignoring errors, let the linux infrastructure handle it.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16262
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

sch-m commented:
I can't remember why I commented out this returns, but this reset controls are not optional.

neheb commented:
Looks like it’s because EPROBE_DEFER is not being propagated.

neheb commented:
@sch-m how about now?

sch-m commented:
> @sch-m how about now?

LGTM. We should change that for the ltq-ptm driver as well.

neheb committed:
ltq-ptm: propagate reset errors to probe

Instead of avoiding returning, propagate error so that the kernel
infrastructure can handle it.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16262
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

neheb commented:
Why did this get closed. Anyway, fixed.

This driver is quite ancient.

neheb commented:
ping @robimarko
